app/models/manageiq/providers/azure/network_manager/refresh_parser.rb

```diff
@@ -0,0 +1,255 @@
+# TODO: Separate collection from parsing (perhaps collecting in parallel a la RHEVM)
+
+class ManageIQ::Providers::Azure::NetworkManager::RefreshParser
+  include ManageIQ::Providers::Azure::RefreshHelperMethods
+  include Vmdb::Logging
+
+  VALID_LOCATION = /\w+/
+  TYPE_DEPLOYMENT = "microsoft.resources/deployments".freeze
+
+  def self.ems_inv_to_hashes(ems, options = nil)
+    new(ems, options).ems_inv_to_hashes
+  end
+
+  def initialize(ems, options = nil)
+    @ems               = ems
+    @config            = ems.connect
+    @subscription_id   = @config.subscription_id
+    @rgs               = ::Azure::Armrest::ResourceGroupService.new(@config)
+    @vns               = ::Azure::Armrest::Network::VirtualNetworkService.new(@config)
+    @ips               = ::Azure::Armrest::Network::IpAddressService.new(@config)
+    @nis               = ::Azure::Armrest::Network::NetworkInterfaceService.new(@config)
+    @nsg               = ::Azure::Armrest::Network::NetworkSecurityGroupService.new(@config)
+    @options           = options || {}
+    @data              = {}
+    @data_index        = {}
+    @resource_to_stack = {}
+  end
+
+  def ems_inv_to_hashes
+    log_header = "Collecting data for EMS : [#{@ems.name}] id: [#{@ems.id}]"
+
+    _log.info("#{log_header}...")
+    get_security_groups
+    get_cloud_networks
+    get_network_ports
+    get_floating_ips
+    _log.info("#{log_header}...Complete")
+
+    @data
+  end
+
+  private
+
+  def parent_manager_fetch_path(collection, ems_ref)
+    @parent_manager_data ||= {}
+    return @parent_manager_data.fetch_path(collection, ems_ref) if @parent_manager_data.has_key_path?(collection,
+                                                                                                      ems_ref)
+
+    @parent_manager_data.store_path(collection,
+                                    ems_ref,
+                                    @ems.public_send(collection).try(:where, :ems_ref => ems_ref).try(:first))
+  end
+
+  def resource_id_for_instance_id(id)
+    # TODO(lsmola) we really need to get rid of the building our own emf_ref, it makes crosslinking impossible, parsing
+    # the id string like this is bad
+    # "/subscriptions/{guid}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/vm1"
+    return nil unless id
+    parts = id.split("/")
+    resource_uid(parts[2],
+                 parts[4].downcase,
+                 parts[6].downcase + "/" + parts[7].downcase,
+                 parts[8])
+  end
```

Comment:
@Ladas -
maybe a more descriptive way of writing this could be:
`_, _, descriptive_var, _, another_descriptive_var, _, another, and_another = id.downcase.split("/")`
this would mean you could remove the example.
Not sure about the negative comment either :sunny: